openapiart/common.py

```diff
@@ -398,8 +419,11 @@ def _set_property(self, name, value, choice=None):
             self._set_choice(name)
             self._properties[name] = self._DEFAULTS[name]
         else:
-            self._set_choice(name)
-            self._properties[name] = value
+            if not self._is_enum_valid(name, value):
+                self._append_error("{} is not a valid enum for property {}".format(value, name))
+            else:
+                self._set_choice(name)
+                self._properties[name] = value
         if self._parent is not None and self._choice is not None and value is not None:
```

Comment:
though I have no ready use case. But I feel we need to add another condition when self._is_enum_valid() will return True. Otherwise we will silently modify that enum. Please ignore this comment if existing condition is sufficient.

Reply:
I feel the existing logic will suffice